Maple2.Server.Game/Model/Field/Actor/ActorState/AnimationState.cs

```diff
@@ -0,0 +1,220 @@
+        public float start;
+        public float end;
+
+        public LoopData(float start, float end) {
+            this.start = start;
+            this.end = end;
+        }
+    }
+
+    public struct TickPair {
+        public long server;
+        public long client;
+
+        // mobs
+        public TickPair(long server) {
+            this.server = server;
+            this.client = server;
+        }
+
+        // players
+        public TickPair(long server, long client) {
+            this.server = server;
+            this.client = client;
+        }
+    }
+
+    public AnimationMetadata? RigMetadata { get; init; }
+    public AnimationSequence? PlayingSequence { get; private set; }
+    private bool IsPlayerAnimation { get; set; }
+    public float MoveSpeed { get; set; }
+    public float AttackSpeed { get; set; }
+    private float sequenceSpeed { get; set; }
+    private float lastSequenceTime { get; set; }
+    private float sequenceEnd { get; set; }
+    private LoopData sequenceLoop { get; set; }
+    private long sequenceEndTick { get; set; }
+    private long sequenceLoopEndTick { get; set; }
+    private long lastTick { get; set; }
+    private bool isLooping { get; set; }
+    private bool loopOnlyOnce { get; set; }
+    private AnimationType sequenceType { get; set; }
+
+    private bool debugPrintAnimations;
+    public bool DebugPrintAnimations {
+        get { return debugPrintAnimations; }
+        set {
+            if (actor is FieldPlayer) {
+                debugPrintAnimations = value;
+            }
+        }
+    }
+
+    public AnimationState(IActor actor, string modelName) {
+        this.actor = actor;
+
+        RigMetadata = actor.NpcMetadata.GetAnimation(modelName);
+        MoveSpeed = 1;
+        AttackSpeed = 1;
+        IsPlayerAnimation = actor is FieldPlayer;
+    }
+
+    private void ResetSequence() {
+        PlayingSequence = null;
+        sequenceSpeed = 1;
+        lastSequenceTime = 0;
+        sequenceLoop = new LoopData(0, 0);
+        lastTick =0;
+        isLooping = false;
+        sequenceEnd = 0;
+        sequenceType = AnimationType.Misc;
+    }
+
+    public bool TryPlaySequence(string name, float speed, AnimationType type) {
+        if (RigMetadata is null) {
+            return false;
+        }
+
+        bool found = RigMetadata.Sequences.TryGetValue(name, out AnimationSequence? sequence);
+
+        ResetSequence();
+
+        if (!found) {
+            DebugPrint($"Attempt to play nonexistent sequence '{name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{sequenceSpeed}");
+
+            return false;
+        }
+
+        DebugPrint($"Playing sequence '{sequence!.Name}' at x{speed} speed, previous: '{PlayingSequence?.Name ?? "none"}' x{sequenceSpeed}");
+
+        PlayingSequence = sequence;
+        sequenceSpeed = speed;
+        sequenceType = type;
+
+        long fieldTick = actor.Field.FieldTick;
+
+        lastTick = fieldTick;
```

Comment:
```suggestion
        lastTick = actor.Field.FieldTick;
```

Reply:
Good catch, thats a leftover from the change lol